evap/evaluation/tests/tools.py

```diff
@@ -63,10 +66,22 @@ def create_course_with_responsible_and_editor(course_id=None):
     contributor = mommy.make(UserProfile, username='responsible')
     editor = mommy.make(UserProfile, username='editor')
 
+    in_one_hour = (timezone.now() + timedelta(hours=1)).replace(second=0, microsecond=0)
+    tomorrow = (timezone.now() + timedelta(days=1)).date
+    course_params = dict(
+        state='prepared',
+        degrees=[mommy.make(Degree)],
+        vote_start_datetime=in_one_hour,
+        vote_end_date=tomorrow
```

Comment:
@karyon My idea here was to make sure that the data generated for tests is realistic: It is not possible to create a Course with with a `vote_start_datetime` in the future through the app, so why should a factory by default violate this validation logic? There are indeed some other tests (e.g. `test_contributor_course_edit) that faced the same problem. Is there any case in which `timezone.now()` is actually a useful value for `vote_start_datetime`?
The same goes for second and microsecond: it is not possible to set them to a value different than `0` through the app.
If you still feel like it would be better to revert these changes and put the logic into the TestCase, I would also be fine with that.

Reply:
thanks. my idea was to not specify anything that the tests do not rely on. if we start adding code that in the end is not needed for the tests to do their thing, that's noise that makes the signal harder to see and understand. e.g. the microsecond thing, our code has no reason to not handle that correctly. if in the future it does not handle that anymore, then it's 100% reasonable to add it, but now it's just additional code that a programmer new to the project might wonder whether it has any significance.

but your point about realistic data is valid, too. i guess we need someone more senior to decide on that :D on a more serious note, i think i still prefer "my" approach, but yours is fine as well, so it's your call. 

note that while contributors must set the start date to the future, and the end date after the start date, staff members do not have the former restriction, only the latter. so they can set both the start and end date into the past. also, when anyone sets the dates, then waits for these dates to pass, and then opens the course again, they would have a form with both dates in the past. the only thing that should really not be possible is an end date that's smaller than the start date. `timezone.now()` might be useful if you want to create an evaluation that should start immediately.

Reply:
just to put it into context: my motivation for this change was that [this test](https://github.com/fsr-itse/EvaP/pull/1269/files/99cf84e234e63647664b3fea84c9a2a69747a17e#diff-ac9ab9135a427edac9993bc57dd13dceR243) failed: the form always submits a datetime with seconds=0 and microseconds=0, so the form submission would always trigger a change, even though we are only approving the course (and the form submission would generally fail, because `mommy`s default `vote_start_datetime = timezone.now()` is already in the past when we do the request)